server/src/main/java/io/crate/expression/scalar/SubscriptFunction.java

```diff
@@ -129,11 +130,11 @@ public static void register(ScalarFunctionModule module) {
 
     private final Signature signature;
     private final Signature boundSignature;
-    private final BiFunction<Object, Object, Object> lookup;
+    private final TriFunction<Object, Object, Boolean, Object> lookup;
```

Comment:
Might be worth introducing a dedicated interface

Reply:
could you clarify the purpose? it is for performance or readability?

Reply:
Mostly readability, but it is a minor thing - fine to keep as well.